### PR TITLE
update troubleshooting for using PSP with User authentication

### PR DESCRIPTION
**Pull request by haouc**
*Issue #, if available:*
#134 
*Description of changes:*
EKS recently switched authentication from service account to User for VPC resource controller. This could be a breaking change. We need call this out in our doc. A PR was also created to update [amazon-eks-user-guide](https://github.com/awsdocs/amazon-eks-user-guide/pull/612)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -230,12 +230,15 @@ If the Pod is still stuck in `ContainerCreating` you can,
...
 ### PSP Blocking Controller Annotations
-If you have a PSP that blocks annotation to Pod, you will have to allow annotation from the following Service Account
+If you have a PSP that blocks annotation to Pod, you will have to allow annotation from the following User `eks:vpc-resource-controller`
 ```
 subjects:
   - kind: Group
     apiGroup: rbac.authorization.k8s.io
     name: system:authenticated
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
```
Comment (jdn5126) [inline docs/troubleshooting.md:240]: Shouldn't the order be switched here? i.e.
```
- kind: User
  apiGroup: rbac.authorization.k8s.io
  name: eks:vpc-resource-controller
```

Comment (haouc) [reply docs/troubleshooting.md:240]: The order should also work. @jdn5126. 
I can change it to be better aligned with documented format.
https://kubernetes.io/docs/reference/access-authn-authz/rbac/

**Review by haouc — COMMENTED** [review]
Change the order to be aligned with documented way.

**Review by jdn5126 — APPROVED** [review]

**Pull request merged**